.github/workflows/pr-check.yml

```diff
@@ -17,8 +17,18 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, ubuntu-latest]
+        auth_type: [sql_login, aad_password, service_principal] # Unique for each parallel job run, part of TEST_DB name to ensure no collisions from parallel jobs
+        include:
+          # Includes the connection string Github secret name, effectively a switch statement on auth_type
+          - auth_type: sql_login
+            connection_string_secret: AZURE_SQL_CONNECTION_STRING_NO_DATABASE
```

Comment:
I think in a future PR I'll rework this so each part of the connection string is its own secret and we build the connection string at runtime. Right now you'll just have to trust that I didn't specify the same connection string 3 times 😆